pkg/backend/rest/server.go

```diff
@@ -75,20 +92,24 @@ func createEchoServer(dbHandler databaseTypes.Database, uiSitePath string, uiBac
 	// Create a router group for the backend /api base URL
 	apiGroup := e.Group(BaseURL)
 
-	// Use oapi-codegen validation middleware to validate
-	// the API group against the OpenAPI schema.
-	apiGroup.Use(middleware.OapiRequestValidator(swagger))
+	// Use oapi-codegen validation middleware to validate the API group against the
+	// OpenAPI schema along with IAM provider.
+	apiGroup.Use(middleware.OapiRequestValidatorWithOptions(swagger, &middleware.Options{
+		Options: openapi3filter.Options{
+			AuthenticationFunc: iam.OapiFilterForProvider(iamProvider),
```

Comment:
Only if "config.IamEnabled" we should set the Authentication middleware

Reply:
Yes. However, in this particular configuration, when `config.IamEnabled = false`, there will be no authentication function (actually, there will, but it will do nothing).